packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -25,6 +25,9 @@ export class TableRow<
     @attr({ attribute: 'record-id' })
     public recordId?: string;
 
+    // The template doesn't bind to `dataRecord`, but making it observable
+    // causes all the template bindings to be re-evaluated when the value changes
+    // because it is part of the binding context for the template.
```

Comment:
The issue with having `cellState` be an accessor on the cell is that something would still need to tell cell that the row's record had been updated.

The approach that I went with is:
- Get rid of `getCellState()` on the row
- Create a `@volatile` property on the row named `columnStates` that returns an array of column + cell state. The implementation of this property is very similar to the implementation of `getCellState()`, but it creates the states for all columns
- Bind the row's template to the new `columnStates` property instead of to the `columns` property

Reply:
lgtm 🚢-it!